tests/utils/test_train_utils.py

```diff
@@ -151,3 +154,34 @@ def test_update_confidence_type(
 ):
     component_config = train_utils.update_confidence_type(component_config)
     assert component_config[MODEL_CONFIDENCE] == model_confidence
+
+
+@pytest.mark.parametrize(
+    "component_config",
+    [
+        ({CHECKPOINT_MODEL: True, EVAL_NUM_EPOCHS: -1, EVAL_NUM_EXAMPLES: 0}),
+        ({CHECKPOINT_MODEL: True, EVAL_NUM_EPOCHS: -1, EVAL_NUM_EXAMPLES: 10}),
+        ({CHECKPOINT_MODEL: True, EVAL_NUM_EPOCHS: 10, EVAL_NUM_EXAMPLES: 0}),
+    ],
+)
+def test_warning_incorrect_checkpoint_setting(component_config: Dict[Text, Text]):
```

Comment:
Given that we check the warning, does it make sense to also check that we don't actually save checkpoints in this case?

Reply:
I *think* this should now be covered by the DIET and TED tests, both of which check that nothing is saved when there are incorrect settings, and also check that the warnings are produced. This may be overkill.. or more dependency on the train_utils method than desired. What do you think?

Reply:
Agree with you.